src/napari_epitools/main.py

```diff
@@ -0,0 +1,381 @@
+    labels: napari.layers.Labels,
+) -> None:
+    """Get filename for exporting Labels cell statitics to CSV"""
+
+    app = magicgui.application.use_app()
+    show_file_dialog = app.get_obj("show_file_dialog")
+    filename = show_file_dialog(
+        mode=FileDialogMode.OPTIONAL_FILE,
+        caption="Specify file to save Epitools cell statistics",
+        start_path=None,
+        filter="*.csv",
+    )
+
+    if filename is None:
+        message = f"Cel statistics not saved for {labels.name} - no filename given"
+        napari.utils.notifications.show_info(message)
+        return
+
+    _cell_statistics_to_csv(
+        filename=filename,
+        labels=labels,
+    )
+
+
+def _cell_statistics_to_csv(
+    filename: os.PathLike,
+    labels: napari.layers.Labels,
+) -> None:
+    """Write cell statistics for all frames in the selected Labels to CSV"""
+
+    try:
+        cell_statistics = labels.metadata["cell_statistics"]
+    except KeyError:
+        message = f"'{labels.name}' has no cell statistics to export"
+        napari.utils.notifications.show_error(message)
+        return
+
+    df = pd.concat(
+        [pd.DataFrame.from_dict(stats).set_index("index") for stats in cell_statistics],
+        keys=list(range(len(cell_statistics))),
+    )
+    df.to_csv(filename, index_label=["frame", "label"])
+
+    # confirm export
+    message = f"'{labels.name}' cell statistics written to '{filename}'"
+    napari.utils.notifications.show_info(message)
+
+
+def create_colourmaps(
+    labels: napari.layers.Labels,
+    colourmap_statistic: str,
+    lower_limit: Widget,
+    upper_limit: Widget,
+    *,
+    autolimits: bool,
+) -> None:
+    """Create a colourmap for a cell statistic each frame.
+
+    If the statistic is set to "id", then the colour mode is set to "auto", which will
+    colour regions based on their ids.
+
+    If the statistics is set to any other value, a colourmap is created for that
+    statistic for each frame.
+    """
+
+    if "cell_statistics" not in labels.metadata:
+        _msg = (
+            "Cannot create colourmaps - "
+            "the selected Labels layer has no cell staistics."
+        )
+        napari.utils.notifications.show_error(_msg)
+        return
+
+    if colourmap_statistic == "id":
+        labels.metadata.pop("metadata", None)
+        labels.color_mode = "auto"
+        return
+
+    cell_statistics = [
+        frame_statistics[colourmap_statistic]
+        for frame_statistics in labels.metadata["cell_statistics"]
+    ]
+    cell_indices = [
+        frame_statistics["index"]
+        for frame_statistics in labels.metadata["cell_statistics"]
+    ]
+
+    if autolimits:
+        lower_limit.value = min(min(statistics) for statistics in cell_statistics)
+        upper_limit.value = max(max(statistics) for statistics in cell_statistics)
+
+    colourmaps = [
+        _create_colourmap(
+            float(lower_limit.value),
+            float(upper_limit.value),
```

Comment:
My IDE is complaining about the four `.value` statements here

>"Widget" has no attribute "value"  [attr-defined]

Not sure how to easily check why this is the case or whether it is okay to ignore this

Reply:
hmm, that's strange. I've changed the annotation to be more precise about the widget type, and now it doesn't seem to complain